### ISSUE TITLE
[TERMINAL_RELEASED_NO_NEW_D05_DEFECT] SWARM-1639 — D05 resume tokenizer/run-root crossbind

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#1639]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SWARM_WORKER_ID: SWARM-1639
STATUS: ACTIVE_NONOWNING_PREQUAL_PENDING_POSTCLAIM_DIRECT_ARBITRATION
SWARM_LANE_KEY: D05|PR590-RESUME|DATA_SOURCE_OR_PIPELINE|TOKENIZER-RUN-ROOT-CROSSBIND-V1
PREFERRED_LANE: #6 D05 — Checkpoints / Reproducibility / Serialization / Conversion
PREFERRED_WORK_KIND: DATA_SOURCE_OR_PIPELINE
SELECTED_WORK_KIND: NONOWNING SOURCE/EXECUTABLE PREQUALIFICATION / D04→D05 IDENTITY CROSSBIND
PARENT_ISSUES: #5 #6 #13 #548 #590 #723 #1256 #1336 #1594 #1599 #1604 #1609 #1615 #1624 #1637 #1638
LOCAL_FREE_ONLY: true
LIVE_MAIN_AT_CLAIM: 58711afeadf7f29c107188cb15d1e73c728768bb
TARGET_PR: #590
TARGET_HEAD_AT_CLAIM: 88be8dfdc490fe496b123c488f269a62e87fbb29
TARGET_BASE_AT_CLAIM: 58711afeadf7f29c107188cb15d1e73c728768bb
TARGET_PRODUCT_OWNER: #1609 / lawful successor
CURRENT_INTEGRATION_OWNER: #1594 / lawful successor
CLAIMED_AT_UTC: THIS_ISSUE_UPDATED_AT_TIMESTAMP_IS_AUTHORITY
LEASE_UNTIL_UTC: CLAIMED_AT_UTC_PLUS_6_HOURS unless terminal/released sooner
PRODUCT_MUTATION: false
AUTHORITATIVE_PR_AUDIT_VERDICT: false
MERGE_AUTHORITY: false
TRAINING_CLAIM: false

ROUTING / COLLISION DECISION:
- SWARM-1639 deterministically maps to permanent lane #6 D05 and work kind DATA_SOURCE_OR_PIPELINE.
- Central integration belongs #1594; PR590 Product/current-main convergence belongs #1609; PR590→external-trainer recovery composition belongs #1615; checkpoint-output addressing finding belongs released #1624→#1609; D05→D06 retention belongs #1637; PR174→PR1336 durable attempt-lineage composition belongs #1638; readiness/tokenizer-decision Product belongs #1599. This package mutates none of them.
- Exact preclaim searches for `D05-RESUME-TOKENIZER-TRUST-DIRECTION` and `expected_tokenizer_hash resume` returned no open claimant for this exact semantic.
- PR590 is still OPEN/non-draft on exact 88be8dfd... against exact base/current main 58711afe....

SOURCE TRUTH BEFORE CLAIM:
1. PR590 checkpoint identity durably stores `tokenizer_hash` and `tokenizer_vocab_hash`; canonical save-side `bind_checkpoint_identity()` also cross-checks run-manifest tokenizer config/vocab/version against supplied tokenizer identity and hashes the complete run manifest.
2. Canonical exported progress-aware `load_trainer_checkpoint()` accepts `expected_tokenizer_hash`, `expected_tokenizer_vocab_hash` and `expected_run_manifest_hash`, but all three are caller-optional.
3. D04 resume binding separately authenticates only ledger/materialization/packing/exposure-plan/ordered-next-exposure identities; it exposes no tokenizer expectation.
4. The D04 double-pack authority independently carries `tokenizer_identity_sha256` plus packing/materialization/ledger identities. Therefore the binary question is whether canonical packing/run authority already transitively and *independently* closes tokenizer identity at D05 restore, or whether a self-consistent checkpoint from another tokenizer/run universe can be restored when callers provide only D04 expectations.
5. Current PR1336 runtime is FRESH_START-only at the inspected live head and therefore cannot be used to pretend this resume boundary is already exercised end-to-end.

BINARY QUESTION / OBJECTIVE AFTER ARBITRATION WIN:
On exact PR590@88be8dfd and current main@58711afe, is canonical fresh-process resume fail-closed against a checkpoint from a different tokenizer/run universe when D04 ledger/materialization/packing/exposure identities remain otherwise expected? Closure requires that an independently trusted exact full run-manifest root, or independently trusted exact tokenizer config+vocab roots, is mechanically mandatory before any model/trainer/RNG mutation. If the loader can restore under D04 expectations while tokenizer/run expectations are omitted, determine whether canonical D04 packing identity itself makes such a substitution impossible; if not, publish one exact HIGH_PRE_STEP same-lineage repair contract to #1609. Do not create checkpoint architecture #2.

OWNED DIAGNOSTIC SURFACE ONLY:
- read-only exact PR590/current-main source + tests around checkpoint run binding, D04 resume binding, progress-aware trainer restore and D04 packing/tokenizer identity;
- LOCAL_FREE isolated executable adversarial reproducer/tests only; no Product commit, branch or PR;
- prove the independent trust direction, not merely checkpoint self-consistency;
- hand one exact closure or concrete finding to #1609 and relevant downstream owner(s), then release immediately.

STRICT NONOVERLAP / AVOID:
- no Product mutation of PR590, PR1256, PR1336, PR1593, PR174 or D04 corpus/tokenizer/packing Product;
- no formal audit PASS/CHANGES_REQUIRED for PR590; #1609 owns Product and must release exact head for a distinct audit semantic;
- no external-trainer recovery composition (#1615), checkpoint locator/content-address repair (#1624), D05→D06 selection/retention (#1637), durable attempt lineage (#1638), readiness tokenizer decision (#1599), or central integration (#1594);
- no optimizer/model update while AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0;
- no paid compute, foreign pretrained weights, external LLM/API corpus intelligence, or final-test outcome access.

PACKAGE_DIMENSIONS:
- independent exact-source verification
- LOCAL_FREE executable adversarial reproducer
- adversarial/substitution trust-direction analysis
- live authority binding
- fresh-process recovery/reproducibility proof
- end-to-end D04→D05 identity-crossbind proof
- operator/Product handoff

ACCEPTANCE:
1. Mandatory postclaim direct OPEN issue collection newest-first at 100/page and separate OPEN PR collection newest-first at 100/page both reach exact empty sentinels; any earlier valid same-semantic claimant wins and SWARM-1639 releases before authoritative finding.
2. Late-bind live main, PR590 exact head/base/state, #1609 Product owner and #1594 integrator immediately before finding; head movement invalidates SHA-bound result until rebound.
3. Trace exact save-side run/tokenizer binding and exact restore-side trust roots; distinguish candidate self-hash from externally expected identity.
4. LOCAL_FREE adversarial proof should attempt two canonical checkpoint/run identities that differ in tokenizer config/vocab/run root while preserving the D04 expectations supplied to restore. If canonical production invariants make that construction impossible, prove the exact transitive hash/authority chain instead of assuming it.
5. Missing/wrong independently expected tokenizer/run root must reject before model, trainer, optimizer/scheduler/scaler, counters or RNG mutation if this crossbind is required for terminal learned-20M recovery.
6. If a gap exists, minimum repair must stay on existing PR590/D05 lineage: make canonical resume consume the already-existing independently authenticated run/tokenizer authority; no second checkpoint or tokenizer framework.
7. This package grants zero training/exposure/learned-weight credit.
8. If binding positive optimized-target exposure + terminal launch/runtime/recovery appears first and no exact training owner exists, release this lower-priority diagnostic and start genuine bounded LOCAL_FREE learned-target training.

TRUTH_BOUNDARY_AT_CLAIM:
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
TOKENIZER_FIT_AUTHORIZED=false
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false
PRODUCT_MUTATION=false
MERGE_AUTHORITY=false

NEXT NOW:
Exhaust mandatory postclaim direct OPEN issue + OPEN PR collections to exact empty sentinels and race-refresh exact semantic. WIN => run exact source/executable trust-direction proof immediately; publish closure/finding and release.

**Comment by Oleksii-debug** [comment]
POSTCLAIM DIRECT ARBITRATION — WIN

- Exact claimed lane: `D05|PR590-RESUME|DATA_SOURCE_OR_PIPELINE|TOKENIZER-RUN-ROOT-CROSSBIND-V1`.
- Direct OPEN issue collection enumerated newest-first at 100/page: pages 1..11 populated; page 12 exact `[]`.
- Separate direct OPEN PR collection enumerated newest-first at 100/page: pages 1..5 populated; page 6 exact `[]`.
- Race-refresh exact-key search returns only SWARM-1639.
- Newer surfaced work (#1640/#1641) owns different D06/D04 semantics, not this D05 resume trust-direction semantic.
- Exact PR590 late-bind remains OPEN/non-draft on `88be8dfdc490fe496b123c488f269a62e87fbb29` with base/current-main `58711afeadf7f29c107188cb15d1e73c728768bb`.
- #1609 remains ACTIVE Product owner for PR590 convergence; SWARM-1639 stays nonowning/read-only toward Product.

DISPOSITION: arbitration WIN. Proceed with LOCAL_FREE isolated source/executable proof only. No Product mutation, no formal PR PASS/CHANGES_REQUIRED, no merge/training authority.

**Comment by Oleksii-debug** [comment]
TERMINAL NONOWNING PREQUAL VERDICT — NO_NEW_D05_DEFECT / CALLER-TRUST REQUIREMENT CONFIRMED / RELEASE

Frozen exact evidence at verdict:
- main `58711afeadf7f29c107188cb15d1e73c728768bb`;
- PR590 exact `88be8dfdc490fe496b123c488f269a62e87fbb29`, OPEN/non-draft, base exact current main;
- Product owner #1609 remains live; runtime Product #1604 remains live; integration #1594 / lawful successor;
- SWARM-1639 arbitration WIN was exhausted before this diagnostic: OPEN issues page 12=`[]`, separate OPEN PRs page 6=`[]`, exact-key race returned only #1639.

RESULT:
The suspected D05 tokenizer/run-root gap is NOT a new same-lineage PR590 Product defect. It is a real trust-direction property of the generic checkpoint API, but the learned-20M terminal contract already requires the missing independent caller root and that requirement is already handed to runtime ownership. Do not add checkpoint architecture #2 or make tokenizer expectations globally mandatory for generic/legacy checkpoint callers.

EXACT SOURCE PROOF:
1. PR590 `CheckpointIdentity` stores `tokenizer_hash`, `tokenizer_vocab_hash`, and complete `run_manifest_hash`. `load_trainer_checkpoint()`/low-level identity checking supports explicit expected tokenizer config/vocab roots and expected full run-manifest root, but those expectations are caller-optional for generic/legacy compatibility.
2. PR590 D04 resume binding carries exactly five D04 SHA roots: ledger, materialization, packing, exposure-plan, ordered-next-exposure. It intentionally does NOT carry tokenizer identity or the full deterministic-double-pack proof root.
3. Canonical D04 materialization identity is tokenizer-bound, and the D04 double-pack proof separately binds `tokenizer_identity_sha256` + materialization + packing. However D05 sees the D04 SHA roots as values supplied by the run manifest; `bind_d04_resume_identity()` does not independently prove that the materialization root was produced under checkpoint tokenizer config/vocab.
4. Canonical packing `identity_sha256` is the packing config hash, not a tokenizer cross-root. Therefore D04 `packing_identity_sha256` alone cannot close tokenizer substitution.
5. PR590 regression `test_complete_d04_binding_validates_without_caller_expectations()` explicitly confirms that a complete D04 binding is structurally accepted without external expectations. This is compatible with the documented generic serializer/load boundary; it must not be mistaken for learned-20M scientific trust.
6. Isolated LOCAL_FREE exact-source-aligned adversarial harness (not a full repo pytest run) constructed a self-consistent checkpoint/run statement with tokenizer B while retaining D04 roots A. D04-only expectations accepted the statement; supplying independently expected tokenizer-A config/vocab roots rejected it. This demonstrates why candidate self-consistency is insufficient and why the existing external-root contract matters.
7. Current canonical D10 launch-input authority already independently cross-binds tokenizer identity with materialization, packing, unique-loss ledger and deterministic-double-pack proof. Current portable run/readiness also carries independently terminal tokenizer authority. Thus an existing trusted upstream root exists; no new authority family is needed.
8. Most importantly, #1609’s already-terminal D04→D05 prequalification explicitly handed #1604 the learned-20M RESUME contract: require externally authenticated checkpoint/run-manifest plus D04 exposure-plan/ordered-next expectations. A genuinely independent exact `expected_run_manifest_hash` is sufficient to reject the mixed tokenizer-B/D04-A checkpoint because PR590 hashes the complete canonical run manifest after save-side tokenizer config/vocab checks. Supplying exact independent tokenizer config+vocab roots as defense-in-depth is also valid, but not necessary to invent as a second trust system.
9. Current PR1336 learned-target runtime is still FRESH_START-only on the inspected lineage; there is no authorized learned-20M RESUME execution path to credit or exploit. Future RESUME qualification remains a runtime/integration consumption problem already owned by #1604/#1594, not a reason to mutate PR590 generic API here.

SCIENTIFIC ACCEPTANCE TO PRESERVE AT FUTURE RESUME BOUNDARY:
- derive `expected_run_manifest_hash` from an independently authenticated canonical run/launch/recovery authority, never from the candidate checkpoint being loaded;
- pass it to D05 restore together with the independently expected D04 resume roots already required by the handoff;
- reject before model/trainer/optimizer/RNG mutation if candidate checkpoint run-manifest root differs;
- if terminal composition cannot independently source the exact run-manifest root, then it MUST instead supply independently trusted tokenizer config+vocab roots (plus D04 roots) or remain blocked; candidate self-hash is not authority;
- add/retain an adversarial terminal-composition regression equivalent to tokenizer-B + D04-A coherent reseal and prove rejection at the caller boundary. This belongs in the existing #1604/#1594 consumption line, not a new D05 framework.

WHY NO NEW HIGH:
A HIGH would be justified if the canonical learned-20M RESUME caller currently omitted the independent run/tokenizer root and nevertheless claimed resume qualification. That caller is not terminal/authorized today; current runtime is FRESH_START-only, and the extant #1609 handoff already requires independent run-manifest trust for RESUME. Therefore the correct action is preserve/consume that contract, not duplicate a known requirement as a new PR590 defect.

OWNERSHIP_RELEASED=true
PRODUCT_MUTATION=false
AUTHORITATIVE_PR_AUDIT_VERDICT=false
MERGE_AUTHORITY=false
TRAINING_CLAIM=false
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
TOKENIZER_FIT_AUTHORIZED=false
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false
